games/xworld/maps/xworld_env.py

```diff
@@ -123,18 +124,73 @@ def set_dims(self, h, w):
         random.shuffle(self.available_grids)
         self.changed = True
 
-    def set_entity(self, type, loc=None, name=None):
+    def set_entity(self, type, loc=None, name=None, force_occupy=False):
         """
-        Add an entity of type to loc which must be currently empty
+        Add an entity instance of type to loc which must be currently empty if 
+        force_occupy is False. If force_occupy is True, then omit the location
+        based availability check.
         """
         if not loc is None:
-            assert loc in self.available_grids, \
-                "set_dims correctly before setting a location"
-            self.available_grids.remove(loc)
+            if not force_occupy:
+                assert loc in self.available_grids, \
+                    "set_dims correctly before setting a location"
+            if loc in self.available_grids:
+                self.available_grids.remove(loc)
         self.entity_nums[type] += 1
         self.entities.append(Entity(type=type, loc=loc, name=name))
         self.changed = True
 
+    def reinstantiate_entity_property(self, entity, property=None, value=None):
+        """
```

Comment:
Is it possible to replace the last two arguments with a dictionary where the key is the property? In this way, one call can change multiple propery-value pairs.

Reply:
done